sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/CosmosFactory.java

```diff
@@ -26,70 +21,73 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CosmosFactory.class);
 
-    private final CosmosConfig config;
+    private final CosmosAsyncClient cosmosAsyncClient;
 
-    private static final boolean IS_TELEMETRY_ALLOWED =
-        PropertyLoader.isApplicationTelemetryAllowed();
+    private final String databaseName;
 
     private static final String USER_AGENT_SUFFIX =
         Constants.USER_AGENT_SUFFIX + PropertyLoader.getProjectVersion();
 
-    private String getUserAgentSuffix() {
+    private static String getUserAgentSuffix() {
         String suffix = ";" + USER_AGENT_SUFFIX;
-
-        if (IS_TELEMETRY_ALLOWED || config.isAllowTelemetry()) {
-            suffix += ";" + MacAddress.getHashMac();
-        }
-
         return suffix;
     }
 
     /**
      * Validate config and initialization
      *
-     * @param cosmosConfig cosmosConfig
+     * @param cosmosAsyncClient cosmosAsyncClient
+     * @param databaseName databaseName
      */
-    public CosmosFactory(@NonNull CosmosConfig cosmosConfig) {
-        validateConfig(cosmosConfig);
+    public CosmosFactory(CosmosAsyncClient cosmosAsyncClient, String databaseName) {
+        Assert.notNull(cosmosAsyncClient, "cosmosAsyncClient must not be null!");
+        Assert.notNull(databaseName, "databaseName must not be null!");
 
-        this.config = cosmosConfig;
+        this.cosmosAsyncClient = cosmosAsyncClient;
+        this.databaseName = databaseName;
     }
 
     /**
      * To create a CosmosAsyncClient
      *
-     * @return CosmosClient
+     * @return CosmosAsyncClient
      */
-    public CosmosAsyncClient getCosmosAsyncClient() {
-        final CosmosClientBuilder cosmosClientBuilderFromConfig =
-            getCosmosClientBuilderFromConfig(config);
-        return cosmosClientBuilderFromConfig.buildAsyncClient();
+    public CosmosAsyncClient getCosmosClient() {
```

Comment:
Rename this method to `getCosmosAsyncClient()`

Reply:
updated